### DISCUSSION TITLE
Cannot access whitelisted directories [solved: add noblacklist for them]

### DISCUSSION THREAD

**Discussion by Tus1688** [netblue30/firejail#4195] [Q&A] [answered]
Hi, I made `thunderbird.profile` in `.config/firejail`, and type following: 

`include /etc/firejail/thunderbird.profile`
`whitelist ${HOME}/Documents/upload`
`whitelist ${HOME}/Pictures/upload`

I did a same think for firefox, when I checked firefox, I could access both whitelisted directories,
But, not with thunderbird. the  Documents & Pictures directories were showed up
![1](https://user-images.githubusercontent.com/76937231/114814625-c0549000-9dde-11eb-869a-2ac551826e00.png)
 but when I click that directory I got following prompt:
![2](https://user-images.githubusercontent.com/76937231/114814653-cfd3d900-9dde-11eb-9eed-b0d833931ad0.png)

I don't have app armor profile enabled by default for thunderbird. This is  the snippet  from `sudo aa-status`:
```
apparmor module is loaded.
22 profiles are loaded.
20 profiles are in enforce mode.
   /usr/bin/man
   /usr/lib/NetworkManager/nm-dhcp-client.action
   /usr/lib/NetworkManager/nm-dhcp-helper
   /usr/lib/connman/scripts/dhclient-script
   /usr/lib/cups/backend/cups-pdf
   /usr/sbin/cups-browsed
   /usr/sbin/cupsd
   /usr/sbin/cupsd//third_party
   /usr/sbin/tcpdump
   /{,usr/}sbin/dhclient
   firejail-default
   ippusbxd
   libreoffice-senddoc
   libreoffice-soffice//gpg
   libreoffice-xpdfimport
   lsb_release
   man_filter
   man_groff
   nvidia_modprobe
   nvidia_modprobe//kmod
2 profiles are in complain mode.
   libreoffice-oopslash
   libreoffice-soffice
9 processes have profiles defined.
9 processes are in enforce mode.
   /usr/sbin/cups-browsed (644) 
   /usr/sbin/cupsd (558) 
   /usr/lib/thunderbird/thunderbird (12309) firejail-default
   /usr/lib/firefox/firefox (12828) firejail-default
   /usr/lib/firefox/firefox (12903) firejail-default
   /usr/lib/firefox/firefox (12956) firejail-default
   /usr/lib/firefox/firefox (13010) firejail-default
   /usr/lib/firefox/firefox (13064) firejail-default
   /usr/lib/firefox/firefox (13110) firejail-default
0 processes are in complain mode.
0 processes are unconfined but have a profile defined.
```
Thanks in advance for your kindly help.

**Comment by Tus1688** [comment]
sorry for unreadable snippets:

apparmor module is loaded.
22 profiles are loaded.
20 profiles are in enforce mode.
   /usr/bin/man
   /usr/lib/NetworkManager/nm-dhcp-client.action
   /usr/lib/NetworkManager/nm-dhcp-helper
   /usr/lib/connman/scripts/dhclient-script
   /usr/lib/cups/backend/cups-pdf
   /usr/sbin/cups-browsed
   /usr/sbin/cupsd
   /usr/sbin/cupsd//third_party
   /usr/sbin/tcpdump
   /{,usr/}sbin/dhclient
   firejail-default
   ippusbxd
   libreoffice-senddoc
   libreoffice-soffice//gpg
   libreoffice-xpdfimport
   lsb_release
   man_filter
   man_groff
   nvidia_modprobe
   nvidia_modprobe//kmod
2 profiles are in complain mode.
   libreoffice-oopslash
   libreoffice-soffice
9 processes have profiles defined.
9 processes are in enforce mode.
   /usr/sbin/cups-browsed (644) 
   /usr/sbin/cupsd (558) 
   /usr/lib/thunderbird/thunderbird (12309) firejail-default
   /usr/lib/firefox/firefox (12828) firejail-default
   /usr/lib/firefox/firefox (12903) firejail-default
   /usr/lib/firefox/firefox (12956) firejail-default
   /usr/lib/firefox/firefox (13010) firejail-default
   /usr/lib/firefox/firefox (13064) firejail-default
   /usr/lib/firefox/firefox (13110) firejail-default
0 processes are in complain mode.
0 processes are unconfined but have a profile defined.

**Comment by rusty-snake** [comment, marked as answer]
> Hi, I made thunderbird.profile in .config/firejail, and type following:
include /etc/firejail/thunderbird.profile
whitelist ${HOME}/Documents/upload
whitelist ${HOME}/Pictures/upload

Name it `thunderbird.local` and skip the `include /etc/firejail/thunderbird.profile`, unless you use firejail < 0.9.58.

> Permission denied

Because they are blacklisted by `include disable-xdg.inc`, add the following:

```
noblacklist ${DOCUMENTS}
noblacklist ${PICTURES}
```

> sorry for unreadable snippets:

Source:

````
Inline `code` in markdown

```
Multi line
code
in markdown
```
````

Rendered:

Inline `code` in markdown

```
Multi line
code
in markdown
```

**Reply by Tus1688** [reply]
hi, thank you for your answer and advice in MD.
